content/en/docs/concepts/services-networking/_index.md

```diff
@@ -7,58 +7,90 @@ description: >
+    the service can change over time.
+
+      * Kubernetes creates
+        [`EndpointSlice`](content/en/docs/concepts/services-networking/endpoint-slices/)
+        objects to provide information about the pods currently
+        backing a Service.
+
+      * A service proxy implementation, takes care of converting the
+        `Service` and `EndpointSlice` objects into packet-rewriting
+        rules using appropriate operating system APIs.
+
+  * The [`Gateway`](/docs/concepts/services-networking/gateway/) API
+    (or its predecessor,
+    [`Ingress`](/docs/concepts/services-networking/ingress/)) allows
+    you to make Services accessible from outside the cluster.
+
+      * A simpler, but less-configurable, mechanism for cluster
+        ingress is available via the Service API's [`type:
+        LoadBalancer`](/docs/concepts/services-networking/service/#loadbalancer),
+        when using a supported {{< glossary_tooltip term_id="cloud-provider">}}.
+
+  * [`NetworkPolicy`](/docs/concepts/services-networking/network-policies)
+    allows you to control traffic between pods, or between pods and
+    the outside world.
+
+In older container systems, there was no automatic connectivity
+between containers on different hosts, and so it was often necessary
+to explicitly create links between containers, or to map container
+ports to host ports to make them reachable by containers on other
+hosts. This is not needed in Kubernetes; Kubernetes's model is that
+`Pods` can be treated much like VMs or physical hosts from the
+perspectives of port allocation, naming, service discovery, load
+balancing, application configuration, and migration.
+
+Only a few parts of this model are implemented by Kubernetes itself.
+For the other parts, Kubernetes defines the APIs, but the
+corresponding functionality is provided by external components:
+
+  * Pod network namespace setup is handled by system-level software
+    implementing the [Container Runtime
+    Interface](/docs/concepts/architecture/cri.md).
+
+  * The pod network itself is managed by a [pod network
+    implementation](/docs/concepts/cluster-administration/addons/#networking-and-network-policy).
+    On Linux, most container runtimes use the
+    {{< glossary_tooltip text="Container Networking Interface (CNI)" term_id="cni" >}}
+    to interact with the pod network implementation, so these
+    implementations are often called "CNI plugins"
+
+  * Kubernetes provides a default implementation of service proxying,
+    called {{< glossary_tooltip term_id="kube-proxy">}}, but some pod
+    network implementations instead use their own service proxy that
+    is more tightly integrated with the rest of the implementation.
+
+  * There are many [implementations of the Gateway
+    API](https://gateway-api.sigs.k8s.io/implementations/), some of
+    which are specific to particular cloud environments, some more
+    focused on "bare metal" environments, and others more generic.
 
```

Comment:
Please add a (repeated) disclaimer / admonition about NetworkPolicy and note that
it may not be enforced even though you can define one.

(aside: wouldn't it be great if there was a default warning, that gets turned off when a network plugin marks NetworkPolicy - the API - as enforced?)

Also:

```suggestion

## {{% heading "whatsnext" %}}

```

Reply:
> (aside: wouldn't it be great if there was a default warning, that gets turned off when a network plugin marks NetworkPolicy - the API - as enforced?)

ah, but then all existing plugins would erroneously trigger the warning until they got updated :slightly_frowning_face:. But we learned from our mistakes at least; AdminNetworkPolicy is a CRD that is installed by the ANP implementation, so if the API is present, then it's implemented. (And the plan, hopefully, is to eventually add "DeveloperNetworkPolicy" as a replacement for NetworkPolicy, which would work the same way, and then we can deprecate NetworkPolicy, and then we'll all have jetpacks and flying cars.)

Until then, the existence of https://github.com/kubernetes-sigs/kube-network-policies means there's no good reason to not implement NetworkPolicy these days so hopefully lack-of-NP support will become less common...

Reply:
We still have to let people know about the risk, so long as there is a network plugin that enables packet networking but not NetworkPolicy.